### PR TITLE
Sass 1403: Create company van fuel benefit amount EOY page

### PR DESCRIPTION
**Pull request by joeyymoore**
### Description

- Creating the company van fuel benefit amount EOY page
- Routing from CYA page change link to the page

Add a link to the relevant story in Jira
https://jira.tools.tax.service.gov.uk/browse/SASS-1403

##### Before Reviewing
- [x]  Have you pulled the branch down?
- [x]  Have you assigned yourself to the PR?
- [x]  Have you moved the task to “in review” on JIRA?
- [x]  Have you checked to ensure all dependencies are up to date?
- [x]  Have you checked to ensure its been rebased against the current version of master?

##### Whilst Reviewing
- [x]  Have you run the tests?
- [x]  Have you run the journey tests?
- [x]  Have you looked at the JIRA story to make sure all Acceptance Criteria has been met?

##### After Reviewing
- [x]  Have you checked for merge conflicts or any changes in the current master that may affect the current pull request? i.e. does it need another rebase?
- [x]  Have you checked to make sure there are no builds in the pipeline before you merge?
- [x]  Have you moved the task to “in pipeline” on Jira?

##### Before creating PR
- [x]  Have you run the tests?
- [x]  Have you run the journey tests? (where applicable)
- [x]  Have you addressed warnings where appropriate?
- [x]  Have you rebased against the current version of master?
- [x]  Have you checked code coverage isn’t lower than previously?

##### After PRs been raised
- [ ]  Have you checked the PR Builder passes?

### CODE REVIEW

File: it/controllers/benefits/CompanyVanFuelBenefitsAmountControllerISpec.scala [hunk 1]
```diff
@@ -0,0 +1,606 @@
...
+            result.status shouldBe OK
+          }
+          titleCheck(user.specificExpectedResults.get.expectedTitle)
+          h1Check(user.specificExpectedResults.get.expectedHeading)
+          captionCheck(expectedCaption)
+          textOnPageCheck(get.expectedContent, contentSelector)
+          textOnPageCheck(amountHint, hintTextSelector)
+          textOnPageCheck(poundPrefixText, poundPrefixSelector)
+          inputFieldCheck(amountInputName, inputSelector)
+
```
Comment (ina-tu) [inline it/controllers/benefits/CompanyVanFuelBenefitsAmountControllerISpec.scala:170]: you could add a check for the input field value to show that it's not pre-filled

File: it/controllers/benefits/CompanyVanFuelBenefitsAmountControllerISpec.scala [hunk 2]
```diff
@@ -0,0 +1,606 @@
...
+            dropEmploymentDB()
+            insertCyaData(employmentUserData(isPrior = true, cyaModel("name", hmrc = true, Some(benefits(fullCarVanFuelModel.copy(vanQuestion = Some(false)))))), userRequest)
+            urlGet(url(taxYearEOY), follow = false, welsh = user.isWelsh, headers = Seq(HeaderNames.COOKIE -> playSessionCookies(taxYearEOY)))
+          }
+
+
+          "has an SEE_OTHER status" in {
+            result.status shouldBe SEE_OTHER
+            result.header("location") shouldBe
+              Some(s"/income-through-software/return/employment-income/$taxYearEOY/check-employment-benefits?employmentId=$employmentId")
```
Comment (ina-tu) [inline it/controllers/benefits/CompanyVanFuelBenefitsAmountControllerISpec.scala:296]: you can add a TODO here as a reminder that the url needs to be changed

File: it/controllers/benefits/CompanyVanFuelBenefitsAmountControllerISpec.scala [hunk 3]
```diff
@@ -0,0 +1,606 @@
...
+              dropEmploymentDB()
+              insertCyaData(employmentUserData(isPrior = true, cyaModel("name", hmrc = true, Some(benefits(fullCarVanFuelModel.copy(vanQuestion = Some(false)))))), userRequest)
+              urlPost(url(taxYearEOY), follow=false,
+                welsh = user.isWelsh, headers = Seq(HeaderNames.COOKIE -> playSessionCookies(taxYearEOY)), body = Map("amount" -> "100"))            }
+
+
+            "has an SEE_OTHER status" in {
+              result.status shouldBe SEE_OTHER
+              result.header("location") shouldBe
+                Some(s"/income-through-software/return/employment-income/$taxYearEOY/check-employment-benefits?employmentId=$employmentId")
```
Comment (ina-tu) [inline it/controllers/benefits/CompanyVanFuelBenefitsAmountControllerISpec.scala:567]: same here

**Review by ina-tu — CHANGES_REQUESTED** [review]

File: it/controllers/benefits/CompanyVanFuelBenefitsAmountControllerISpec.scala (hunk 2 above)
Comment (joeyymoore) [reply it/controllers/benefits/CompanyVanFuelBenefitsAmountControllerISpec.scala:296]: The setup and outcome for the test was wrong. Updated it to reflect the scenario so no TODO needed.

**Pull request merged**